R/call.R

```diff
@@ -49,9 +216,13 @@ call_new <- function(f, ..., .args = list()) {
 #'
 #' # Add an explicit missing argument
 #' call_modify(call, list(na.rm = quote(expr = )))
-call_modify <- function(call, new_args, env = parent.frame()) {
-  stopifnot(is.call(call), is.list(new_args))
-
+#'
+#' # If the call is missing, the parent frame is used instead.
+#' f <- function(bool = TRUE) call_modify(new_args = list(bool = FALSE))
+#' f()
+call_modify <- function(call = NULL, new_args, env = NULL) {
```

Comment:
Two options:

1. Added some special behaviour for when `call_modify()` is missing
2. Default `new_args` to `list()`.